app/models/travel_pattern.rb

```diff
@@ -150,4 +150,110 @@ def to_calendar
 
     return calendar
   end
+
```

Comment:
Looks good and I believe it should work as intended.
One thing that might be an improvement is to refactor these methods into scopes.
Ie

`scope :filter_by_purpose, ~> (purpose) { joins(:purposes).merge( Purpose.where(name: purpose) ) unless purpose.blank? }`

Or, at least, something similar to that. I'd say this is optional though.